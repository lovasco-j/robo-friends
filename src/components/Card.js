import React from 'react';

const Card = ({id, name, email: username, company}) => {
    return(
        <div className='bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
            <img src={`https://robohash.org/${id}?size=200x200`} alt="robot"/>
            <h2>{name}</h2>
            <p>{username}</p>
            <p>{company.name}</p>
        </div>
    );
}

export default Card;
