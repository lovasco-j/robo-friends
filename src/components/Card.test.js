import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('<Card> should match the snapshot', () => {
    const mockProps = { id: 0, name: 'Test', email: 'j@l', company: 'rb@friends' }
    expect(shallow(<Card {...mockProps} />)).toMatchSnapshot();
});

