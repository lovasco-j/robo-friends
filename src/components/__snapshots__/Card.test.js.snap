// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    company="rb@friends"
    email="j@l"
    id={0}
    name="Test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robot"
          src="https://robohash.org/0?size=200x200"
/>,
        <h2>
          Test
</h2>,
        <p>
          j@l
</p>,
        <p />,
      ],
      "className": "bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot",
          "src": "https://robohash.org/0?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "j@l",
        },
        "ref": null,
        "rendered": "j@l",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robot"
            src="https://robohash.org/0?size=200x200"
/>,
          <h2>
            Test
</h2>,
          <p>
            j@l
</p>,
          <p />,
        ],
        "className": "bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robot",
            "src": "https://robohash.org/0?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "j@l",
          },
          "ref": null,
          "rendered": "j@l",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
